### PR TITLE
feat: create component BeagleDynamicList to use in ListView and GridView

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Description and Example

This code was based on angular code, Created concept to Dynamic List, and use to generate ListView and GridView.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.
- [X] I have made the documentation changes or I created an issue explaining how to document this change on [Docs issues](https://github.com/ZupIT/beagle-docs/issues). Please link the issue here:

[PR Guide]: https://github.com/ZupIT/beagle/blob/main/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-flutter/blob/ff7f81c6d41cf4ae8b2882ddfd2bc0a159d802ed/beagle/test/bridge_impl/global_context_js_test.dart#L97">beagle/test/bridge_impl/global_context_js_test.dart#L97</a> - The type argument(s) of the function 'get' can't be inferred</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-flutter/blob/ff7f81c6d41cf4ae8b2882ddfd2bc0a159d802ed/beagle/test/bridge_impl/global_context_js_test.dart#L109">beagle/test/bridge_impl/global_context_js_test.dart#L109</a> - The type argument(s) of the function 'get' can't be inferred</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-flutter/blob/ff7f81c6d41cf4ae8b2882ddfd2bc0a159d802ed/beagle_components/lib/src/beagle_screen.dart#L55">beagle_components/lib/src/beagle_screen.dart#L55</a> - 'textTheme' is deprecated and shouldn't be used. This property is no longer used, please use toolbarTextStyle and titleTextStyle instead. This feature was deprecated after v2.4.0-0.0.pre.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_flutter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: beagle_components/lib/src/default_components.dart [hunk 1]
```diff
@@ -86,6 +88,53 @@ ComponentBuilder beagleScrollViewBuilder() {
...
+        isScrollIndicatorVisible:
+            element.getAttributeValue('isScrollIndicatorVisible'),
+        scrollEndThreshold: element.getAttributeValue('scrollEndThreshold'),
+        iteratorName: element.getAttributeValue('iteratorName'),
+        identifierItem: element.getAttributeValue('key'),
+        onScrollEnd: element.getAttributeValue('onScrollEnd'),
+        children: children,
+        spanCount: element.getAttributeValue('spanCount'),
+        suffix: element.getAttributeValue('__suffix__'),
+        beagleWidgetStateProvider: BeagleWidgetStateProvider(),
```
Comment (hernandazevedozup) [inline beagle_components/lib/src/default_components.dart:134]: Nice !

File: beagle_components/lib/src/utils/beagle_widget_state_provider.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:beagle/beagle.dart';
+import 'package:flutter/widgets.dart';
+
+class BeagleWidgetStateProvider {
```
Comment (hernandazevedozup) [inline beagle_components/lib/src/utils/beagle_widget_state_provider.dart:20]: Nice implementation, we can now refactor where we need this implementation to use this class. I think we use this also on SimpleForm.

**Review by hernandazevedozup — DISMISSED** [review]

File: beagle_components/lib/src/dynamic_list/beagle_dynamic_list.dart [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    final anchor = _getAnchor();
+
+    final beagleWidgetState = widget.beagleWidgetStateProvider.of(context);
+
+    beagleWidgetState.view.getRenderer().doTemplateRender(
+          templateManager: templateManager,
+          anchor: anchor,
+          contexts: contexts,
+          componentManager: _iterateComponent,
+          mode: null,
```
Comment (arthurbleilzup) [inline beagle_components/lib/src/dynamic_list/beagle_dynamic_list.dart:167]: you do not need to set the `mode`, if you want to set, please use `TreeUpdateMode.replace`

**Review by arthurbleilzup — CHANGES_REQUESTED** [review]

**Review by arthurbleilzup — APPROVED** [review]

**Pull request merged**